src/main/java/com/aws/greengrass/clientdevices/auth/session/MqttSessionFactory.java

```diff
@@ -54,19 +46,17 @@ public Session createSession(Map<String, String> credentialMap) throws Authentic
     }
 
     private Session createIotThingSession(MqttCredential mqttCredential) throws AuthenticationException {
-        try {
-            Optional<Certificate> cert = certificateRegistry.getCertificateFromPem(mqttCredential.certificatePem);
-            if (!cert.isPresent() || !cert.get().isActive()) {
-                throw new AuthenticationException("Certificate isn't active");
-            }
-            Thing thing = thingRegistry.getOrCreateThing(mqttCredential.clientId);
-            if (!thingRegistry.isThingAttachedToCertificate(thing, cert.get())) {
-                throw new AuthenticationException("unable to authenticate device");
-            }
-            return new SessionImpl(cert.get(), thing);
-        } catch (CloudServiceInteractionException | InvalidCertificateException e) {
-            throw new AuthenticationException("Failed to verify certificate with cloud", e);
+        // NOTE: We should remove  calling this useCase from here, but for now it serves its purpose. We will
+        //  refactor this later
+        CreateIoTThingSession useCase = useCases.get(CreateIoTThingSession.class);
+        CreateSessionDTO command = new CreateSessionDTO(mqttCredential.clientId, mqttCredential.certificatePem);
+        Result<Session> sessionResult = useCase.apply(command);
+
+        if (sessionResult.isOk()) {
+            return sessionResult.get();
         }
+
+        throw new AuthenticationException(sessionResult.getError());
```

Comment:
I know we've gone back and forth about this functional style, but masking exceptions like this violates our Greengrass coding guidelines.